Makefile

```diff
@@ -33,16 +33,27 @@ lint:  ## Lint project source files
 	golint -set_exit_status ./...
 
 .PHONY: test
-test:  ## Run unit tests
+test: test-unit test-integration-http test-integration-websocket  ## Run all tests
+
+.PHONY: test-unit
+test-unit:  ## Run unit tests
 	@ # Note: this requires go1.10+ in order to do multi-package coverage reports
 	go test -short -race -coverprofile=coverage.out -covermode=atomic ./...
 
-.PHONY: test-integration
-test-integration:  ## Run integration tests
+.PHONY: test-integration-http
+test-integration-http:  ## Run integration tests for http client
+	-docker-compose -f compose/server.yml rm -fsv
+	docker-compose -f compose/server.yml up -d
+	sleep 6
+	go test -race -cover -run IntegrationHTTP ./... || (docker-compose -f compose/server.yml stop; exit 1)
+	docker-compose -f compose/server.yml down
+
+.PHONY: test-integration-websocket
+test-integration-websocket:  ## Run integration tests for websocket client
```

Comment:
Why do we run the http and websocket integration tests separately? I'm sure there's a reason, just wanted to know

Reply:
We have a custom config for the emulator plugin to run the tests against. If we let the http and websocket integration tests run together, the http ones would go first and modify the config. Once the websocket ones run, we have to run the tests upon the modified config. It's confusing sometimes. So basically, letting them run separately makes sure that they are independent of each other and alway have a fresh start.